projects/packages/publicize/src/class-keyring-service-helper.php

```diff
@@ -0,0 +1,341 @@
+
+		if ( $service ) {
+			$url = add_query_arg( array( 'service' => $service ), $url );
+		}
+
+		if ( count( $params ) ) {
+			$url = add_query_arg( $params, $url );
+		}
+
+		return $url;
+	}
+
+	/**
+	 * Build a connection URL (sharing settings page with unique query args to create a connection).
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $service_name Service name.
+	 * @param string $for          Feature name.
+	 */
+	public static function connect_url( $service_name, $for ) {
+		return add_query_arg(
+			array(
+				'action'   => 'request',
+				'service'  => $service_name,
+				'kr_nonce' => wp_create_nonce( 'keyring-request' ),
+				'nonce'    => wp_create_nonce( "keyring-request-$service_name" ),
+				'for'      => $for,
+			),
+			admin_url( 'options-general.php?page=sharing' )
+		);
+	}
+
+	/**
+	 * Build a URL to refresh a connection (sharing settings page with unique query args to refresh a connection).
+	 * Similar to connect_url, but with a refresh parameter.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $service_name Service name.
+	 * @param string $for          Feature name.
+	 */
+	public static function refresh_url( $service_name, $for ) {
+		return add_query_arg(
+			array(
+				'action'   => 'request',
+				'service'  => $service_name,
+				'kr_nonce' => wp_create_nonce( 'keyring-request' ),
+				'refresh'  => 1,
+				'for'      => $for,
+				'nonce'    => wp_create_nonce( "keyring-request-$service_name" ),
+			),
+			admin_url( 'options-general.php?page=sharing' )
+		);
+	}
+
+	/**
+	 * Build a URL to delete a connection (sharing settings page with unique query args to delete a connection).
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $service_name Service name.
+	 * @param string $id           Connection ID.
+	 */
+	public static function disconnect_url( $service_name, $id ) {
+		return add_query_arg(
+			array(
+				'action'   => 'delete',
+				'service'  => $service_name,
+				'id'       => $id,
+				'kr_nonce' => wp_create_nonce( 'keyring-request' ),
+				'nonce'    => wp_create_nonce( "keyring-request-$service_name" ),
+			),
+			admin_url( 'options-general.php?page=sharing' )
+		);
+	}
+
+	/**
+	 * Build contents handling Keyring connection management into Sharing settings screen.
+	 *
+	 * @since $$next-version$$
+	 */
+	public static function admin_page_load() {
+		if ( isset( $_GET['action'] ) ) {
+			if ( isset( $_GET['service'] ) ) {
+				$service_name = $_GET['service']; // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- We verify below.
+			}
+
+			switch ( $_GET['action'] ) {
+
+				case 'request':
+					check_admin_referer( 'keyring-request', 'kr_nonce' );
+					check_admin_referer( "keyring-request-$service_name", 'nonce' );
+
+					$verification = ( new Secrets() )->generate( 'publicize' );
+					if ( ! $verification ) {
+						$url = Jetpack::admin_url( 'jetpack#/settings' );
```

Comment:
Since the Jetpack class doesn't exist in the Social plugin, I think this will be a problem.

Reply:
I handled this similarly to above, by commenting out the action and then adding a comment.